edx_proctoring/static/proctoring/js/views/proctored_exam_view.js

```diff
@@ -0,0 +1,130 @@
+            this.listenTo(this.model, 'change', this.modelChanged);
+
+            $(window).unbind('beforeunload', this.unloadMessage);
+
+            /* make the async call to the backend REST API */
+            /* after it loads, the listenTo event will file and */
+            /* will call into the rendering */
+            this.model.fetch();
+        },
+        modelChanged: function () {
+            // if we are a proctored exam, then we need to alert user that he/she
+            // should not be navigating around the courseware
+            var taking_as_proctored = this.model.get('taking_as_proctored');
+            var time_left = this.model.get('time_remaining_seconds') > 0;
+            var status = this.model.get('attempt_status');
+            var in_courseware = document.location.href.indexOf('/courses/' + this.model.get('course_id') + '/courseware/') > -1;
+
+            if ( taking_as_proctored && time_left && in_courseware && status !== 'started'){
+                $(window).bind('beforeunload', this.unloadMessage);
+            } else {
+                // remove callback on unload event
+                $(window).unbind('beforeunload', this.unloadMessage);
+            }
+
+            this.render();
+        },
+        render: function () {
+            if (this.template !== null) {
+                if (
+                    this.model.get('in_timed_exam') &&
+                    this.model.get('time_remaining_seconds') > 0 &&
+                    this.model.get('attempt_status') !== 'error'
+                ) {
+                    var html = this.template(this.model.toJSON());
+                    this.$el.html(html);
+                    this.$el.show();
+                    this.updateRemainingTime(this);
+                    this.timerId = setInterval(this.updateRemainingTime, 1000, this);
+
+                    // Bind a click handler to the exam controls
+                    var self = this;
+                    $('.exam-button-turn-in-exam').click(function(){
+                        $(window).unbind('beforeunload', self.unloadMessage);
+
+                        $.ajax({
+                            url: '/api/edx_proctoring/v1/proctored_exam/attempt/' + self.model.get('attempt_id'),
+                            type: 'PUT',
+                            data: {
+                              action: 'stop'
+                            },
+                            success: function() {
+                              // Reloading page will reflect the new state of the attempt
+                              location.reload();
+                            }
+                        });
+                    });
+                    //$('.proctored-exam-action-stop').css('cursor', 'pointer');
+                }
+            }
+            return this;
+        },
+        unloadMessage: function  () {
+            return gettext("Are you sure you want to leave this page? \n" +
```

Comment:
@chrisndodge Is this the message that's shown when students leave the exam but are still navigating within the course in the existing edX browser window?
@griffresch Since this message text was drafted the behavior has changed; it's no longer true that students have to submit their responses within the time limit. 
Suggested revised text:
"Are you sure you want to leave this page?
If you leave the exam, the timer continues counting down. You can return to your exam, but you will have lost the time that you spent away from it."